### PR TITLE
pluginhandler: use uname from the host

### PR DESCRIPTION
**Pull request by kyrofa**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [x] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh unit`?

-----

Currently, when recording the manifest from the `PluginHandler`, the snapcraft CLI runs `uname` using `common.run_output`, which is made for running binaries out of parts or the staging area. That's never desired in this case, so this PR fixes LP: [#1780825](https://bugs.launchpad.net/snapcraft/+bug/1780825) by using `subprocess` directly to always run `uname` from the host.

### CODE REVIEW

File: snapcraft/internal/pluginhandler/__init__.py [hunk 1]
```diff
@@ -569,8 +570,11 @@ def mark_build_done(self):
             scriptlet_metadata=self._scriptlet_metadata[steps.BUILD]))
 
     def _get_machine_manifest(self):
+        # Use subprocess directly here. common.run_output will use binaries out
+        # of the snap, and we want to use the one on the host.
+        uname = subprocess.check_output(['uname', '-srvmpio'])
```
Comment (sergiusens) [inline snapcraft/internal/pluginhandler/__init__.py]: We will need a specific error for this now (it was catched by the general wrapper until this change).

File: snapcraft/internal/pluginhandler/__init__.py [hunk 2]
```diff
@@ -569,8 +570,11 @@ def mark_build_done(self):
...
         return {
-            'uname': common.run_output(['uname', '-srvmpio']),
+            'uname': uname.decode(sys.getfilesystemencoding()).strip(),
```
Comment (sergiusens) [inline snapcraft/internal/pluginhandler/__init__.py]: We will also need to catch unicode decode errors and use a surrogate on errors.

Comment (kyrofa) [reply snapcraft/internal/pluginhandler/__init__.py]: Not sure how uname would cause this, but seems harmless. I added the extra case.

**Comment by kyrofa** [comment]
> My suggestion would be to create a module/class for uname that is self contained (like the multipass module/class).

That feels a bit premature/overkill for running a single command in a single location. I opted for simply making it more robust in-place, let me know if you still want it extracted.

File: snapcraft/internal/pluginhandler/__init__.py [hunk 3]
```diff
@@ -569,8 +570,24 @@ def mark_build_done(self):
             scriptlet_metadata=self._scriptlet_metadata[steps.BUILD]))
 
     def _get_machine_manifest(self):
+        # Use subprocess directly here. common.run_output will use binaries out
+        # of the snap, and we want to use the one on the host.
+        try:
+            output = subprocess.check_output(['uname', '-srvmpio'])
```
Comment (sergiusens) [inline snapcraft/internal/pluginhandler/__init__.py]: mind using uname's long options for clarity and purpose

Comment (kyrofa) [reply snapcraft/internal/pluginhandler/__init__.py]: Sure, done.

**Comment by sergiusens** [comment]
Thank you for the added clarity. My network is bonkers, but tests seem to fail, feel free to land if no major changes take place to fix them :smile:

**Pull request merged**